test/e2e/throughputanomalydetection_test.go

```diff
@@ -166,78 +168,156 @@ func testAnomalyDetectionDelete(t *testing.T, data *TestData, connect *sql.DB) {
-			"1.550": "true",
-			"9.755": "true"},
+			"4.004": "true",
+			"4.005": "true",
+			"4.006": "true",
+			"5.000": "true",
+			"2.002": "true",
+			"2.003": "true",
+			"2.500": "true",
+		},
 		"DBSCAN": {
 			"1.000": "true",
 			"1.005": "true",
 			"5.000": "true",
 			"3.260": "true",
-			"2.058": "true"},
+			"2.058": "true",
+			"5.002": "true",
+			"5.027": "true",
+			"2.500": "true",
+			"1.029": "true",
+			"1.630": "true"},
 	}
-	for _, algo := range algoNames {
-		_, jobName, err := tadrunJob(t, data, algo)
-		require.NoError(t, err)
-		err = data.podWaitForReady(defaultTimeout, jobName+"-driver", flowVisibilityNamespace)
-		require.NoError(t, err)
-		err = waitTADJobComplete(t, data, jobName, tadjobCompleteTimeout)
-		require.NoErrorf(t, err, "Throughput Anomaly Detection Spark job failed to complete")
-		stdout, err := tadretrieveJobResult(t, data, jobName)
-		resultArray := strings.Split(stdout, "\n")
-		assert := assert.New(t)
-		length := len(resultArray)
-		assert.GreaterOrEqualf(length, 3, "stdout: %s", stdout)
-		assert.Containsf(stdout, "throughput", "stdout: %s", stdout)
-		assert.Containsf(stdout, "algoCalc", "stdout: %s", stdout)
-		assert.Containsf(stdout, "anomaly", "stdout: %s", stdout)
-
-		for i := 1; i < length; i++ {
-			// check metrics' value
-			resultArray[i] = strings.TrimSpace(resultArray[i])
-			if resultArray[i] != "" {
-				resultArray[i] = strings.ReplaceAll(resultArray[i], "\t", " ")
-				tadoutputArray := strings.Fields(resultArray[i])
-				anomaly_output := tadoutputArray[11]
-				assert.Equal(12, len(tadoutputArray), "tadoutputArray: %s", tadoutputArray)
-				switch algo {
-				case "ARIMA":
-					algo_throughput := tadoutputArray[10][:5]
-					assert.Equal(result_map["ARIMA"][algo_throughput], anomaly_output, "Anomaly outputs dont match in tadoutputArray: %s", tadoutputArray)
-				case "EWMA":
-					algo_throughput := tadoutputArray[10][:5]
-					assert.Equal(result_map["EWMA"][algo_throughput], anomaly_output, "Anomaly outputs dont match in tadoutputArray: %s", tadoutputArray)
-				case "DBSCAN":
-					throughput := tadoutputArray[7][:5]
-					assert.Equal(result_map["DBSCAN"][throughput], anomaly_output, "Anomaly outputs dont match in tadoutputArray: %s", tadoutputArray)
-				}
+	assert_variable_map := map[string]map[string]int{
+		"None": {
+			"tadoutputArray_len": 12,
+			"anomaly_output_idx": 11,
+			"throughput_idx":     7},
+		"podName": {
+			"tadoutputArray_len": 10,
+			"anomaly_output_idx": 9,
+			"throughput_idx":     5},
+		"podLabel": {
+			"tadoutputArray_len": 11,
+			"anomaly_output_idx": 10,
+			"throughput_idx":     6},
+		"external": {
+			"tadoutputArray_len": 8,
+			"anomaly_output_idx": 7,
+			"throughput_idx":     3},
+		"svc": {
+			"tadoutputArray_len": 8,
+			"anomaly_output_idx": 7,
+			"throughput_idx":     3},
+	}
+	poolIdx = 0
+	for agg_type, algoName := range aggTypeToAlgoNameMap {
+		poolIdx += 1
+		if agg_type == "None" {
+			for _, algo := range algoNames {
```

Comment:
Thanks for addressing this. I was thinking one randomly selected algorithm is good enough for the "None" agg_type when I wrote the last comment, not a strong opinion if you think it's better to test all algorithms for the "None" agg_type.

Reply:
Yes, we would like to keep all the tests for "None" case as was before the Agg TAD.